mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+        'survival_prob_fit_err': survival_prob_fit_err,
+        'amplitude_fit': amplitude_fit,
+        'amplitude_fit_err': amplitude_fit_err,
+        'data_hist_freq': data_hist_freq,
+        'reduced_chi2': reduced_chi2
+        }
+        return dictionary_of_fit_results
+
+    #simulated resolution with scatter_proportion floating, without reconstruction eff curve, without detection eff curve
+    def make_spectrum_ftc_2(self, prob_parameter, scatter_proportion, emitted_peak='shake'):
+        gases = self.gases
+        current_path = self.path_to_scatter_spectra_file
+        # check_existence_of_scatter_files()
+        #filenames = list_files('scatter_spectra_files')
+        p = np.zeros(len(gases))
+        p[0:-1] = scatter_proportion
+        p[-1] = 1 - sum(scatter_proportion)
+        scatter_spectra_file_path = os.path.join(current_path, 'scatter_spectra.npy')
+        scatter_spectra = np.load(
+        scatter_spectra_file_path, allow_pickle = True
+        )
+        en_array = self.std_eV_array()
+        current_full_spectrum = np.zeros(len(en_array))
+        emitted_peak = self.base_shape
+        if emitted_peak == 'lorentzian':
+            current_working_spectrum = self.std_lorenztian_17keV()
+        elif emitted_peak == 'shake':
+            current_working_spectrum = self.shakeSpectrumClassInstance.shake_spectrum()
+        elif emitted_peak == 'dirac':
+            current_working_spectrum = self.std_dirac()
+
+        if self.use_combined_four_trap_inst_reso:
+            current_working_spectrum = self.convolve_ins_resolution_combining_four_trap(current_working_spectrum, self.trap_weights)
+        else:
+            current_working_spectrum = self.convolve_ins_resolution(current_working_spectrum)
+
+        zeroth_order_peak = current_working_spectrum
+        current_full_spectrum += current_working_spectrum
+        N = len(self.gases)
+        for M in range(1, self.max_scatters + 1):
+            gas_scatter_combinations = np.array([np.array(i) for i in product(range(M+1), repeat=N) if sum(i)==M])
+            for combination in gas_scatter_combinations:
+                entry_str = ''
+                for component, gas_type in zip(combination, self.gases):
+                    entry_str += gas_type
+                    entry_str += str(component).zfill(2)
+                current_working_spectrum = scatter_spectra.item()[entry_str]
+                current_working_spectrum = self.normalize(signal.convolve(zeroth_order_peak, current_working_spectrum, mode='same'))
+                coefficient = factorial(sum(combination))
+                for component, i in zip(combination, range(len(self.gases))):
+                    coefficient = coefficient/factorial(component)*p[i]**component
+                current_full_spectrum += coefficient*current_working_spectrum*prob_parameter**M
+        return current_full_spectrum
+
+    def spectrum_func_ftc_2(self, bins_Hz, *p0):
```

Comment:
add description

Reply:
Added the description of spectrum_func_ftc_2